File: Punto4/hanoi.py
import time

def hanoi(n, from_rod, to_rod, aux_rod):
    if n == 1:
        # print(f"Mover disco de {from_rod} a {to_rod}")
        return
    hanoi(n-1, from_rod, aux_rod, to_rod)
    # print(f"Mover disco de {from_rod} a {to_rod}")
    hanoi(n-1, aux_rod, to_rod, from_rod)

n = 20

inicio = time.time()
hanoi(n, 'A', 'C', 'B')
fin = time.time()

print(f"Torres de Hanoi con {n} discos completado.")
print(f"Tiempo en Python: {fin - inicio} segundos")

File: Punto4/hanoi.c
#include <stdio.h>
#include <time.h>

void hanoi(int n, char from, char to, char aux) {
    if (n == 1) {
        // printf("Mover disco de %c a %c\n", from, to);
        return;
    }
    hanoi(n-1, from, aux, to);
    // printf("Mover disco de %c a %c\n", from, to);
    hanoi(n-1, aux, to, from);
}

int main() {
    int n = 20;
    clock_t start, end;
    double tiempo;

    start = clock();
    hanoi(n, 'A', 'C', 'B');
    end = clock();

    tiempo = ((double)(end - start)) / CLOCKS_PER_SEC;

    printf("Torres de Hanoi con %d discos completado.\n", n);
    printf("Tiempo en C: %f segundos\n", tiempo);

    return 0;
}
